### ISSUE TITLE
Crash while rendering ModelSelect2 widget in 3.3.0rc5

### ISSUE THREAD

**Issue by sdeleeuw** [yourlabs/django-autocomplete-light#986]
I am using Django 1.8.

**What am I doing:**
```
from dal import autocomplete

class MyForm(forms.Form):
    some_field = forms.ModelChoiceField(
        queryset=SomeModel.objects.all(),
        widget=autocomplete.ModelSelect2(url='some_url')
    )
```

**Stacktrace:**
```
...
File "/usr/lib/python2.7/site-packages/django/forms/forms.py" in __str__
  537.         return self.as_widget()
File "/usr/lib/python2.7/site-packages/django/forms/forms.py" in as_widget
  593.         return force_text(widget.render(name, self.value(), attrs=attrs))
File "/usr/lib/python2.7/site-packages/dal/widgets.py" in render
  150.         widget = super(WidgetMixin, self).render(name, value, attrs, renderer)
File "/usr/lib/python2.7/site-packages/django/forms/widgets.py" in render
  513.         options = self.render_options(choices, [value])
File "/usr/lib/python2.7/site-packages/dal/widgets.py" in render_options
  124.         html = super(WidgetMixin, self).render_options(*args)
File "/usr/lib/python2.7/site-packages/django/forms/widgets.py" in render_options
  539.         for option_value, option_label in chain(self.choices, choices):

Exception Type: TypeError at ...
Exception Value: 'NoneType' object is not iterable
```

**Conclusion:**
* Issue is Django version related.
* This doesn't happen in DAL version 3.2.10.
* dal.widgets.WidgetMixin.render calls super() with invalid arguments.

See inspection below.

**Inspection:**
```
>>> from dal_select2.widgets import ModelSelect2
>>> pprint(ModelSelect2.__mro__)
(<class 'dal_select2.widgets.ModelSelect2'>,
 <class 'dal.widgets.QuerySetSelectMixin'>,
 <class 'dal.widgets.WidgetMixin'>,
 <class 'dal_select2.widgets.Select2WidgetMixin'>,
 <class 'django.forms.widgets.Select'>,
 <class 'django.forms.widgets.Widget'>,
 <type 'object'>)

# dal.widgets.WidgetMixin
class WidgetMixin(object):
    ...
    def render(self, name, value, attrs=None, renderer=None):
        """Call Django render together with `render_forward_conf`."""
        # ----> 4th argument is render <----
        widget = super(WidgetMixin, self).render(name, value, attrs, renderer)
        try:
            field_id = attrs['id']
        except (KeyError, TypeError):
            field_id = name
        conf = self.render_forward_conf(field_id)
        return mark_safe(widget + conf)

# django.forms.widgets.Select
class Select(Widget):
    ...
    # ----> 4th argument is choices <----
    def render(self, name, value, attrs=None, choices=()):
        if value is None:
            value = ''
        final_attrs = self.build_attrs(attrs, name=name)
        output = [format_html('<select{}>', flatatt(final_attrs))]
        options = self.render_options(choices, [value])
        if options:
            output.append(options)
        output.append('</select>')
        return mark_safe('\n'.join(output))

```

Signature of django.forms.widgets.Select.render(), the inclusion of the 'rederer' parameter, has changed since Django 1.11.

**Comment by jpic** [comment]
Would love to review a patch for this ;)

**Comment by sdeleeuw** [comment]
Fair enough :-) 
I'll provide one within the next few days, when I found some time for it.

**Comment by sdeleeuw** [comment]
@jpic 
I believe you have already fixed this issue in commit:
https://github.com/yourlabs/django-autocomplete-light/commit/823247eb4a0e1b97ae9bd54e1e7eef05cc8ea868

I just upgraded from 3.3.0rc5 to master branch, and it works okay.
This issue can be closed.

**Comment by jpic** [comment]
Sorry, I thought my fix was in the release that you reported still broken :/

Released master on 3.3.0-rc6

Thanks for your support
